Bug: Parser fails when returning struct literal directly

## Description

Returning a struct literal directly in a return statement causes a parse error. The parser does not recognize struct instantiation syntax (`StructName{...}`) after the `return` keyword.

## Current Behavior

When attempting to return a struct literal directly, the parser fails with an "expected expression" error:

```
error[E2003]: expected expression, found end of block
  --> ./test.ez:8:1
   |
8 | }
   | ^ expected expression
```

## Expected Behavior

Should be able to return struct literals directly without assigning to an intermediate variable:

```ez
const User struct {
    id int
    name string
}

do createUser(id int, name string) -> User {
    return User{id: id, name: name}  // Should work
}
```

## Reproduction Steps

1. Define a struct type
2. Create a function that returns that struct type
3. Try to return a struct literal directly using `return StructName{...}`

**test.ez:**
```ez
module main

const User struct {
    id int
    name string
}

do createUser(id int, name string) -> User {
    return User{id: id, name: name}
}

do main() {
    temp user = createUser(1, "Alice")
}
```

## Workaround

Assign to a temp variable first, then return the variable:

```ez
do createUser(id int, name string) -> User {
    temp u = User{id: id, name: name}
    return u  // This works
}
```

## Possible Fix

The issue is in `parseExpression()` or the struct literal parsing logic. When parsing the return value expression and encountering an `IDENT` followed by `LBRACE`, the parser should recognize this as a struct literal instantiation.

In `pkg/parser/parser.go`, the `parseReturnStatement()` function calls `parseExpression(LOWEST)` to parse return values. The expression parser needs to handle the case where an identifier is followed by `{` to indicate struct instantiation.

Check if `parseIdentifier()` or `parseExpression()` properly handles the `IDENT{...}` pattern for struct literals. The fix likely involves:

1. When parsing an identifier expression, check if peek token is `LBRACE`
2. If so, parse as struct literal: `StructName{field: value, ...}`
3. Return the struct literal expression

Look at how struct literals are parsed in assignment context (`temp u = User{...}`) and ensure the same logic applies in return statement context.

## Environment

- Branch: feat/module-system
- Found: 2025-11-30

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -861,13 +861,10 @@ func (p *Parser) parseBlockStatementWithSuppress(suppressions []*Attribute) *Blo
 			switch stmt.(type) {
 			case *ReturnStatement, *BreakStatement, *ContinueStatement:
 				unreachable = true
-				// If we're at the closing brace after a terminating statement,
-				// don't advance - let the loop exit naturally
-				if p.currentTokenMatches(RBRACE) {
-					continue // Skip p.nextToken() and re-check loop condition
-				}
 			}
 		}
+		// Always advance to the next token after parsing a statement.
+		// The loop condition will exit when we reach the block's closing brace.
 		p.nextToken()
 	}
 

--- a/test/return_struct_literal_test.ez
+++ b/test/return_struct_literal_test.ez
@@ -1,0 +1,48 @@
+import @std
+
+const Point struct {
+    x int
+    y int
+}
+
+const User struct {
+    id int
+    name string
+    active bool
+}
+
+do createPoint() -> Point {
+    return Point{x: 10, y: 20}
+}
+
+do createUser() -> User {
+    return User{id: 1, name: "Alice", active: true}
+}
+
+do createPointInline() -> Point {
+    return Point{x: 5, y: 15}
+}
+
+do main() {
+    using std
+
+    println("=== Return Struct Literal Tests ===")
+    println("")
+
+    println("Test 1: Return struct literal with named fields")
+    temp p Point = createPoint()
+    println("  Point: (${p.x}, ${p.y})")
+
+    println("Test 2: Return struct with multiple field types")
+    temp u User = createUser()
+    println("  User id: ${u.id}")
+    println("  User name: ${u.name}")
+    println("  User active: ${u.active}")
+
+    println("Test 3: Return struct literal inline")
+    temp p2 Point = createPointInline()
+    println("  Point2: (${p2.x}, ${p2.y})")
+
+    println("")
+    println("=== All Return Struct Literal Tests Passed! ===")
+}

